python/ray/util/spark/cluster_init.py

```diff
@@ -720,47 +698,126 @@ def background_job_thread_fn():
+    address to connect to the cluster. To shut down the cluster you can call
+    `ray.util.spark.shutdown_ray_cluster()`.
     Note: If the active ray cluster haven't shut down, you cannot create a new ray
     cluster.
 
-    Args
-        num_worker_nodes: The number of spark worker nodes that the spark job will be
-            submitted to. This argument represents how many concurrent spark tasks will
-            be available in the creation of the ray cluster. The ray cluster's total
-            available resources (memory, CPU and/or GPU) is equal to the quantity of
-            resources allocated within these spark tasks.
-            Specifying the `num_worker_nodes` as `-1` represents a ray cluster
-            configuration that will use all available spark tasks slots (and resources
-            allocated to the spark application) on the spark cluster.
-            To create a spark cluster that is intended to be used exclusively as a
+    Args:
+        num_worker_nodes: This argument represents how many ray worker nodes to start
+            for the ray cluster.
+            Specifying the `num_worker_nodes` as `ray.util.spark.MAX_NUM_WORKER_NODES`
+            represents a ray cluster
+            configuration that will use all available resources configured for the
+            spark application.
+            To create a spark application that is intended to exclusively run a
             shared ray cluster, it is recommended to set this argument to
-            `ray.spark.utils.MAX_NUM_WORKER_NODES`.
+            `ray.util.spark.MAX_NUM_WORKER_NODES`.
+        num_cpus_per_node: Number of cpus available to per-ray worker node, if not
+            provided, use spark application configuration 'spark.task.cpus' instead.
+            **Limitation** Only spark version >= 3.4 or Databricks Runtime 12.x
+            supports setting this argument.
+        num_gpus_per_node: Number of gpus available to per-ray worker node, if not
+            provided, use spark application configuration
+            'spark.task.resource.gpu.amount' instead.
+            This argument is only available on spark cluster that is configured with
+            'gpu' resources.
+            **Limitation** Only spark version >= 3.4 or Databricks Runtime 12.x
+            supports setting this argument.
         object_store_memory_per_node: Object store memory available to per-ray worker
             node, but it is capped by
             "dev_shm_available_size * 0.8 / num_tasks_per_spark_worker".
             The default value equals to
-            "dev_shm_available_size * 0.8 / num_tasks_per_spark_worker".
-        head_options: A dict representing Ray head node options.
-        worker_options: A dict representing Ray worker node options.
+            "0.3 * spark_worker_physical_memory * 0.8 / num_tasks_per_spark_worker".
+        head_node_options: A dict representing Ray head node extra options.
```

Comment:
you probably need to mention  these will be passed as `ray start` options and the conversion rules, otherwise people won't know what values can be set here.